va/validation-authority.go

```diff
@@ -191,6 +191,38 @@ func (va ValidationAuthorityImpl) validateDvsni(identifier core.AcmeIdentifier,
 	return challenge, err
 }
 
+func (va ValidationAuthorityImpl) validateDNS(identifier core.AcmeIdentifier, input core.Challenge) (core.Challenge, error) {
+	challenge := input
+
+	if identifier.Type != core.IdentifierDNS {
+		challenge.Status = core.StatusInvalid
+		err := fmt.Errorf("Identifier type for DNS was not itself DNS")
+		return challenge, err
+	}
+
+	const DNSPrefix = "_acme-challenge"
+
+	challengeSubdomain := fmt.Sprintf("%s.%s", DNSPrefix, identifier.Value)
+	txts, err := net.LookupTXT(challengeSubdomain)
```

Comment:
We should probably use [`miekg/dns`](https://godoc.org/github.com/miekg/dns) here and a method similar to the CAA checking (although luckily minus the manual record parsing) so that we can explicitly define the resolver used, and check for DNSSEC validation failures.